File: src/audio/SimpleItem.java
package audio;

import java.io.File;
import java.net.URI;

/**
 * Simplest {@link Item} implementation. Wraps {@link java.net.URI}. Immutable.
 *
 * @author Martin Polakovic
 */
public class SimpleItem extends Item {

	private final URI uri;

	public SimpleItem(URI _uri) {
		uri = _uri;
	}

	public SimpleItem(File file) {
		uri = file.toURI();
	}

	public SimpleItem(Item i) {
		uri = i.getURI();
	}

	@Override
	public final URI getURI() {
		return uri;
	}

}

File: src/audio/SimpleItem.kt
package audio

import java.io.File
import java.net.URI

/** Simplest [Item] implementation. Wraps [java.net.URI]. Immutable. */
class SimpleItem: Item {

    override val uri: URI

    constructor(resource: URI) {
        uri = resource
    }

    constructor(resource: File) {
        uri = resource.toURI()
    }

    override fun toSimple() = this

}